api/src/controllers/team.js

```diff
@@ -92,21 +92,23 @@ function getTeam(req, res, next) {
  * @param {*} res response object - updated team object
  * @returns 201: returns updated team details
  */
-async function addTeam(req, res, next) {
+function addTeam(req, res, next) {
   Team.findOne({ email: req.body.email })
     .then((foundTeam) => {
       if (foundTeam) {
         return res.status(400).send('Email had been registered');
       }
+      bcrypt.genSalt().then((salt) => 
+        bcrypt.hash(req.body.password, salt).then((hashedPassword) => {
+          const hashedTeam = { ...req.body, password: hashedPassword };
+          Team.create(hashedTeam)
+            .then(() => {
+              return res.status(201).json('Team has been created successfully')
+            })
```

Comment:
curious: why are we going from try-catch style to promise style? Shouldn't the refactor be the other way around to avoid callback hell: https://blog.avenuecode.com/callback-hell-promises-and-async/await?
I'd suggest going back to simply async await, try-catch

Reply:
Sure, you reckon we raise a PR to convert all the promises back to async/await?

Reply:
Yep, that'd be great!